File: cu-manager/src/main/webapp/app/scripts/services/userService.js
/*
 * LICENCE : CloudUnit is available under the Gnu Public License GPL V3 : https://www.gnu.org/licenses/gpl.txt
 *     but CloudUnit is licensed too under a standard commercial license.
 *     Please contact our sales team if you would like to discuss the specifics of our Enterprise license.
 *     If you are not sure whether the GPL is right for you,
 *     you can always test our software under the GPL and inspect the source code before you contact us
 *     about purchasing a commercial license.
 *
 *     LEGAL TERMS : "CloudUnit" is a registered trademark of Treeptik and can't be used to endorse
 *     or promote products derived from this project without prior written permission from Treeptik.
 *     Products or services derived from this software may not be called "CloudUnit"
 *     nor may "Treeptik" or similar confusing terms appear in their names without prior written permission.
 *     For any questions, contact us : contact@treeptik.fr
 */

/**
 * @ngdoc service
 * @name webuiApp.UserService
 * @description # UserService Factory in the webuiApp.
 */


(function () {

  'use strict';

  angular
    .module('webuiApp')
    .factory('UserService', UserService);

  UserService.$inject = [
    '$cookieStore',
    '$http',
    '$interval',
    'ErrorService',
    '$rootScope'
  ];

//////////////////////////////////////////

  function UserService($cookieStore, $http, $interval, ErrorService, $rootScope) {

    var user = {};
    user.check = check;
    user.profile = profile;
    user.logout = logout;
    user.createLocalSession = createLocalSession;
    user.deleteLocalSession = deleteLocalSession;
    user.isLogged = isLogged;
    user.getRole = getRole;
    user.checkUserStatus = checkUserStatus;

    return user;

    function check(username, password) {

      var data, headers;

      data = 'j_username=' + username + '&j_password=' + password + '&submit=Login';

      headers = {
        'Content-Type': 'application/x-www-form-urlencoded',
        'Accept': 'application/json, text/plain, */*'
      };

      return $http.post('/user/authentication', data, {
        headers: headers
      });
    }

    function profile() {
      return $http.get('/user/status', {});
    }

    function logout() {
      deleteLocalSession();
      return $http.get('/user/logout', {});
    }

    function createLocalSession() {
      $rootScope.authenticated = true;
    }

    function deleteLocalSession() {
      $rootScope.authenticated = false;
    }

    function isLogged() {
      console.log("$rootScope.authenticated: " + $rootScope.authenticated);
      return $rootScope.authenticated;
    }

    function getRole() {
      profile().then(function(user){
        return user.data.role.description;
      });
    }

    function checkUserStatus(){
      getUserStatus();
      return $interval(function(){
        getUserStatus();
      }, 2000);
    }

    function getUserStatus() {
      profile()
        .then(function (user) {
          // si le statut est à 3 on gel l'IHM
          if(user.data.status === 3){
            $rootScope.$broadcast(':freezeSystem');
          } else {
            $rootScope.$broadcast(':unFreezeSystem');
          }
        })
        .catch(function (error) {
          ErrorService.handle(error);
        });
    }
  }
})();

File: cu-manager/src/main/java/fr/treeptik/cloudunit/service/UserService.java
/*
 * LICENCE : CloudUnit is available under the GNU Affero General Public License : https://gnu.org/licenses/agpl.html
 * but CloudUnit is licensed too under a standard commercial license.
 * Please contact our sales team if you would like to discuss the specifics of our Enterprise license.
 * If you are not sure whether the AGPL is right for you,
 * you can always test our software under the AGPL and inspect the source code before you contact us
 * about purchasing a commercial license.
 *
 * LEGAL TERMS : "CloudUnit" is a registered trademark of Treeptik and can't be used to endorse
 * or promote products derived from this project without prior written permission from Treeptik.
 * Products or services derived from this software may not be called "CloudUnit"
 * nor may "Treeptik" or similar confusing terms appear in their names without prior written permission.
 * For any questions, contact us : contact@treeptik.fr
 */

package fr.treeptik.cloudunit.service;

import fr.treeptik.cloudunit.exception.CheckException;
import fr.treeptik.cloudunit.exception.ServiceException;
import fr.treeptik.cloudunit.model.User;

import java.util.List;

public interface UserService {

    List<User> findByEmail(String email)
        throws ServiceException;

    List<User> findAll()
        throws ServiceException;

    User findById(Integer id)
        throws ServiceException;

    void remove(User user)
        throws ServiceException, CheckException;

    User update(User user)
        throws ServiceException;

    User create(User user)
        throws ServiceException, CheckException;

    void activationAccount(User user)
        throws ServiceException;

    void changePassword(User user, String newPassword)
        throws ServiceException;

    User findByLogin(String login)
        throws ServiceException;

    void changeEmail(User user, String newEmail)
        throws ServiceException;

    String sendPassword(User user)
        throws ServiceException;

    void deleteAllUsersMessages(User user)
        throws ServiceException;

    void changeUserRights(String login, String roleValue)
        throws ServiceException, CheckException;

}
